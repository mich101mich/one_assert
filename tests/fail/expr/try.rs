fn main() -> Result<(), ()> {
    one_assert::assert!(true?);
    one_assert::assert!(Ok(1)?);

    Ok(())
}
