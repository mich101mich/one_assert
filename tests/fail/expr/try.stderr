error[E0277]: the `?` operator can only be applied to values that implement `Try`
 --> tests/fail/expr/try.rs:2:25
  |
2 |     one_assert::assert!(true?);
  |                         ^^^^^ the `?` operator cannot be applied to type `bool`
  |
  = help: the trait `Try` is not implemented for `bool`

error[E0308]: `?` operator has incompatible types
 --> tests/fail/expr/try.rs:3:25
  |
3 |     one_assert::assert!(Ok(1)?);
  |                         ^^^^^^ expected `bool`, found integer
  |
  = note: `?` operator cannot convert from `{integer}` to `bool`
